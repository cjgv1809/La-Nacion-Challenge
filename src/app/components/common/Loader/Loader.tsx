function Loader() {
  return (
    <div className="loader" role="loader">
      <div className="outer"></div>
      <div className="middle"></div>
      <div className="inner"></div>
    </div>
  );
}

export default Loader;
