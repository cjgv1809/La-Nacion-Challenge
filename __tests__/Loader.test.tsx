import { render, screen } from "@testing-library/react";
import Loader from "@/app/components/common/Loader";

describe("Loader", () => {
  it("renders the loader", () => {
    render(<Loader />);

    const loader = screen.getByRole("loader");

    expect(loader).toBeInTheDocument();
  });

  it("renders the loader with the correct classes", () => {
    render(<Loader />);

    const loader = screen.getByRole("loader");

    expect(loader).toHaveClass("loader");
  });
});
