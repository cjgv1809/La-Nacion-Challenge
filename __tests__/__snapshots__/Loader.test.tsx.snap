// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="loader"
    role="loader"
  >
    <div
      class="outer"
    />
    <div
      class="middle"
    />
    <div
      class="inner"
    />
  </div>
</DocumentFragment>
`;
